src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -58,6 +58,302 @@ const HWIntrinsicInfo& HWIntrinsicInfo::lookup(NamedIntrinsic id)
     return hwIntrinsicInfoArray[id - NI_HW_INTRINSIC_START - 1];
 }
 
+#if defined(TARGET_XARCH)
+const TernaryLogicInfo& TernaryLogicInfo::lookup(uint8_t control)
+{
+    // This table is 768 bytes and is about as small as we can make it.
+    //
+    // The way the constants work is we have three keys:
+    // * A: 0xF0
```

Comment:
curious, how did you come up with the values of these keys? Are they related to the encoding or just something you picked?

Reply:
They are directly part of the encoding and are covered in the architecture manuals.